### ISSUE TITLE
Allow to update PR branch 

### ISSUE THREAD

**Issue by Czaki** [peter-evans/create-pull-request#2061]
### Subject of the issue

In [napari](https://github.com/napari/napari/) we use this PR to autogenerate dependencies pin. We would like to be able to update constraints inside PR and regenerate it using pip-compile, then update PR description and title. 

Without regeneration, this action works really nicely. Still, when trying such a scenario, I must squash all commits as this action only cherrypicks the latest commit instead of using the whole branch history.

Is it possible to use all branches instead of the latest commit? 

Currently working on this in this PR: https://github.com/napari/napari/pull/5904

**Comment by peter-evans** [comment]
Hi @Czaki 

I'm not really sure what you are trying to do, sorry. I suspect you are trying to create the pull request branch before calling the action. Ideally, you should let the action create and update the branch itself.

**Comment by Czaki** [comment]
I will try to explain what I want using draw:

![image](https://github.com/peter-evans/create-pull-request/assets/3826210/8cdc2c63-4c4e-4f83-b50d-21a3140b92de)

I have a repository with commits A,B,C on the main branch. Then automatically, the workflow creates commit D and opens PR. Then I added a commit E, which triggered the workflow again. 
I expect this will create commit F, and the content of all 3 commits will be in PR. But workflow creates commit G with the content expected for commit F, so all changes of commits D and E are lost. 

As the `main` is provided as main branch I do not understand this behavior. 

Is it cleaner now?

**Comment by peter-evans** [comment]
You have two options I think:
1. Create a workflow that combines the changes in commit D and commit E and let the action create/update a PR that always produces a PR with the correct combined changes. (Note, you should not create the branch yourself. Just make the changes in the workflow and let the action create/update the branch. See the docs/examples if you don't understand what I mean.)
2. Use an [alternative strategy](https://github.com/peter-evans/create-pull-request#alternative-strategy---always-create-a-new-pull-request-branch), which creates the PR with commit D but does not update it any further.

**Comment by Czaki** [comment]
I landed with option one (I need to update the title and description). As I understand, adding the option to keep history could be hard.

**Comment by peter-evans** [comment]
> As I understand, adding the option to keep history could be hard.

Yes. The action is designed this way deliberately. There is no commit history because the action treats each run as if it was executing for the first time for consistency. I've commented about this before in other issues. e.g. https://github.com/peter-evans/create-pull-request/issues/1871#issuecomment-1504562985

I'm going to close this issue now because I think we've reached a conclusion that the workflow just needed to be redesigned a bit to fit with how this action works.
